src/2.2.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016 Andrew Vaziri
+// CtCI Question #
+// |||Prompt|||
+// Implment an alogirthm  to find the kith to last element of a singly linekd
+// list.
+// |||Questions I might ask|||
+// --How am I givent he linked list? What exactly should I return?
+// --When we say Kth from last, is that zero based or one based?
+// --What should I do if there are less than K-1 elements in the list? What
+// should I do if someone gives me a negative value for K?
+// --Can I assume the list holds ints for simplicty sake?
+
+// |||Possible Implementations|||
+
+#include <forward_list>
+#include <iterator>
+#include <iostream>
+#include "../include/2.2.h"
+
+namespace CtCI {
+
+// |||Implementation 1|||
+// Iterate through the list once to count the number of elements. Use the
+// result to calculate which element to return. Assuming zero indexed. That
+// would be the (N-k-1)th element after the begining.
```

Comment:
There were three solutions in the book. What you describe is the one I did not implement. Its a fair point.